### ISSUE TITLE
ClassCastException when using CALLS_REAL_METHODS with Varargs and Java 11

### ISSUE THREAD

**Issue by loetifuss** [mockito/mockito#2733]
When using static mocking with CALLS_REAL_METHODS to mock methods containing Varargs parameters Mockito will throw ClassCastExceptions when running the test under Java 11. If the test is run using Java 1.8 no error occurs.

Consider this (simplified) example: 

Class under test:
```
package de;

public class Foo {

	static boolean isBar(String... vars) {
		return false;
	}

}
```
Test class:
```
package de;

import org.junit.jupiter.api.Test;
import org.mockito.MockedStatic;
import org.mockito.Mockito;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

public class FooTest {

	@Test
	void testName() throws Exception {

		try(MockedStatic<Foo> mocked = mockStatic(Foo.class, Mockito.CALLS_REAL_METHODS)) {
			assertFalse(Foo.isBar("one", "two"));

		}

	}

}
```
When run with JDK 11 this fails with the following exception, under JDK 1.8 it works.
Using List instead of Varargs also works under JDK 11.

```
java.lang.ClassCastException: Cannot cast [Ljava.lang.String; to java.lang.String
	at java.base/java.lang.Class.cast(Class.java:3605)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:710)
	at java.base/java.lang.invoke.MethodHandleImpl$AsVarargsCollector.invokeWithArguments(MethodHandleImpl.java:578)
	at de.Foo.isBar(Foo.java:8)
	at de.FooTest.testName(FooTest.java:18)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:675)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
```

Used versions: JDK 1.8.0_281 / Microsoft JDK 11.0.12+7 / Mockito 3.10.0

**Comment by Vividious** [comment]
What I can see, it got fixed in 4.8.0 version.
I have the same exception under JDK 11/Ubuntu/Mockito 3.10 -> ... -> 4.7.0

**Comment by big-andy-coates** [comment]
If this is fixed, can someone close the issue then?
